eth/stagedsync/stage_bor_heimdall.go

```diff
@@ -206,12 +202,14 @@ func BorHeimdallForward(
 		nextSpanId = binary.BigEndian.Uint64(k) + 1
 	}
 	snapshotLastSpanId := cfg.blockReader.(LastFrozen).LastFrozenSpanID()
-	if snapshotLastSpanId+1 > nextSpanId {
+	if s.BlockNumber != 0 && snapshotLastSpanId+1 > nextSpanId {
 		nextSpanId = snapshotLastSpanId + 1
 	}
 	var endSpanID uint64
 	if headNumber > zerothSpanEnd {
 		endSpanID = 2 + (headNumber-zerothSpanEnd)/spanLength
+	} else {
+		endSpanID = 1
```

Comment:
Initially, we can simply set this to 1 so that on the first block itself, it fetches both 0th and 1st span.